shell/shared/grain.js

```diff
@@ -1300,6 +1304,7 @@ Router.map(function () {
     },
 
     onStop: function () {
+      this.state.set("beforeActionHookRan", undefined);
```

Comment:
As far as I know, this is the closest possible approximation to removing an entry from the `ReactiveDict`. We use this pattern elsewhere. https://github.com/sandstorm-io/sandstorm/blob/25796cffb46f9a8c77fb270c92aa7363c113af5e/shell/shared/shell.js#L785

It would work just as well to set it to `false`. I have no strong opinion one way or the other.